packages/engine-server/src/markdown/remark/noteRefsV2.ts

```diff
@@ -845,13 +845,17 @@ function convertNoteRefHelperAST(
   } else {
     // Otherwise, just clone the existing proc instead of creating a new one.
```

Comment:
@SeriousBug, @kevinslin - I'm not as familiar w/ the MD code, but this change may put noteRefProc in a slightly different state than before.  Can you think of any issues with the logic?  I tested in test-workspace, particularly [[Links|dendron.ref.links]], and it looks to be parity in preview + hover from what I can tell.

Another question; zooming out to view the code right above this change:
```typescript
  if (dest === DendronASTDest.HTML) {
    // For HTML, we need to make sure that we don't use a processor with HTML
    // target. Otherwise as we process recursive references, the HTML output
    // from deeper levels is broken (everything gets converted into `<div>`s for
    // some reason)
    noteRefProc = MDUtilsV5.procRemarkFull(
      {
        ...MDUtilsV5.getProcData(proc),
        insideNoteRef: true,
        fname: note.fname,
        vault: note.vault,
        engine,
      },
      MDUtilsV5.getProcOpts(proc)
    );
  } else {
    // Otherwise, just clone the existing proc instead of creating a new one.
    // This will largely preserve existing opts, we just need to change a few.
    noteRefProc = proc();
    // proc is the parser that was parsing the note the reference was in, so need to update fname to reflect that we are parsing the referred note
    MDUtilsV5.setProcData(noteRefProc, {
      fname: note.fname,
      insideNoteRef: true,
      vault: note.vault,
    });
  }
  ```
  
After my change, the logic is actually the same within the if and else branch, so we can collapse it.  But I wanted to check prior to deleting the `// For HTML, we need to make sure...` comment.  How does this work?  I'm not sure what state in `proc` has been altered / is necessary to alter to support the `dest === DendronASTDest.HTML` scenario.

Reply:
Creating a new proc used to cause bugs because some plugins would not be created in the new proc, which is something I had to fix by reverting to this. The issues are generally caused by MDUtilsV4/V5 mismatch, and by any code that attaches a plugin to the processor outside of `procRemark...` functions. I don't quite remember exactly what bug it was fixing though, perhaps it's not a concern anymore if we already refactored those other code that was giving problems.

Although, create a new processor instead of cloning the existing one?

For the code above, the rest of the comment explains it:
```
    // ... Otherwise as we process recursive references, the HTML output
    // from deeper levels is broken (everything gets converted into `<div>`s for
    // some reason)
```
If you do remove it, you should check that nested note references work correctly. You'll likely have to update this branch to override the dest if the dest is `HTML`.

Reply:
I tested this the if/else collapsed and can confirm recursive references work fine

Reply:
feel free to collapse and merge